modules/custom/activity_basics/src/Plugin/ActivityContext/ContentInMyGroupActivityContext.php

```diff
@@ -107,17 +107,16 @@ public function getRecipients(array $data, $last_uid, $limit) {
   /**
    * {@inheritdoc}
    */
-  public function isValidEntity($entity) {
+  public function isValidEntity(EntityInterface $entity) {
     // Check if it's placed in a group (regardless off content type).
-    if ($entity instanceof EntityInterface) {
-      if ($entity->getEntityTypeId() === 'group_content') {
+    if ($entity->getEntityTypeId() === 'group_content') {
+      return TRUE;
+    }
+
+    if ($entity->getEntityTypeId() === 'post') {
+      if (!$entity->field_recipient_group->isEmpty()) {
         return TRUE;
       }
-      elseif ($entity->getEntityTypeId() === 'post') {
-        if (!$entity->field_recipient_group->isEmpty()) {
-          return TRUE;
-        }
-      }
     }
 
     return FALSE;
```

Comment:
This looks like a great case for `switch` :D 

```
switch ($entity->getEntityTypeId()) {
  case 'group_content':
    return TRUE;
  case 'post:
    return !$entity->field_recipient_group->isEmpty();
  default:
    return FALSE;
```

Reply:
Done